From: Mauro Carvalho Chehab <mchehab@redhat.com>
Date: Fri, 12 Oct 2012 16:26:43 -0400
Subject: [PATCH] snd-tea575x: Fix radio-shark build, when PCI is not compiled

On ARM devices, those vars get undefined:

ERROR: "snd_tea575x_init" [drivers/media/radio/radio-shark.ko] undefined!
ERROR: "snd_tea575x_exit" [drivers/media/radio/radio-shark.ko] undefined!

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

diff --git a/sound/Kconfig b/sound/Kconfig
index 261a03c..fef150c 100644
--- a/sound/Kconfig
+++ b/sound/Kconfig
@@ -77,6 +77,12 @@ source "sound/drivers/Kconfig"
 
 source "sound/isa/Kconfig"
 
+# I2C device, used by sound/pci and by some drivers/media USB devices
+config SND_TEA575X
+	tristate
+	depends on SND_FM801_TEA575X_BOOL || SND_ES1968_RADIO || RADIO_SF16FMR2 || RADIO_MAXIRADIO || RADIO_SHARK
+	default y
+
 source "sound/pci/Kconfig"
 
 source "sound/ppc/Kconfig"
diff --git a/sound/pci/Kconfig b/sound/pci/Kconfig
index ff3af6e..f93fda7 100644
--- a/sound/pci/Kconfig
+++ b/sound/pci/Kconfig
@@ -1,10 +1,5 @@
 # ALSA PCI drivers
 
-config SND_TEA575X
-	tristate
-	depends on SND_FM801_TEA575X_BOOL || SND_ES1968_RADIO || RADIO_SF16FMR2 || RADIO_MAXIRADIO
-	default SND_FM801 || SND_ES1968 || RADIO_SF16FMR2 || RADIO_MAXIRADIO
-
 menuconfig SND_PCI
 	bool "PCI sound devices"
 	depends on PCI

From: Arnd Bergmann <arnd@arndb.de>
Subject: [PATCH] ARM: export read_current_timer
	https://patchwork.kernel.org/patch/1361481/

	read_current_timer is used in the get_cycles() function when
	ARM_ARCH_TIMER is set, and that function can be inlined into
	driver modules, so we should export the function to avoid
	errors like

	ERROR: "read_current_timer" [drivers/video/udlfb.ko] undefined!
	ERROR: "read_current_timer" [crypto/tcrypt.ko] undefined!

	Signed-off-by: Arnd Bergmann <arnd@arndb.de>
	Cc: Shinya Kuribayashi <shinya.kuribayashi.px@renesas.com>
	Cc: Stephen Boyd <sboyd@codeaurora.org>
	Cc: Will Deacon <will.deacon@arm.com>
	Cc: Russell King <rmk+kernel@arm.linux.org.uk>

diff --git a/arch/arm/kernel/arch_timer.c b/arch/arm/kernel/arch_timer.c
index cf25880..6327d1f 100644
--- a/arch/arm/kernel/arch_timer.c
+++ b/arch/arm/kernel/arch_timer.c
@@ -14,6 +14,7 @@
 #include <linux/device.h>
 #include <linux/smp.h>
 #include <linux/cpu.h>
+#include <linux/export.h>
 #include <linux/jiffies.h>
 #include <linux/clockchips.h>
 #include <linux/interrupt.h>
@@ -232,6 +233,7 @@ int read_current_timer(unsigned long *timer_val)
 	*timer_val = arch_counter_get_cntpct();
 	return 0;
 }
+EXPORT_SYMBOL_GPL(read_current_timer);
 
 static struct clocksource clocksource_counter = {
 	.name	= "arch_sys_counter",

From: David Cullen <david.cullen@koe-americas.com>
Subject: [PATCH] Fix error: unknown field reclaim_buffers specified in initializer

	The reclaim_buffers field has been removed from struct drm_driver.

	Signed-off-by: David Cullen <david.cullen@koe-americas.com>
diff --git a/drivers/staging/omapdrm/omap_drv.c
b/drivers/staging/omapdrm/omap_drv.c
index 4beab94..44149ee 100644
--- a/drivers/staging/omapdrm/omap_drv.c
+++ b/drivers/staging/omapdrm/omap_drv.c
@@ -761,7 +761,6 @@ static struct drm_driver omap_drm_driver = {
 		.irq_postinstall = dev_irq_postinstall,
 		.irq_uninstall = dev_irq_uninstall,
 		.irq_handler = dev_irq_handler,
-		.reclaim_buffers = drm_core_reclaim_buffers,
 #ifdef CONFIG_DEBUG_FS
 		.debugfs_init = omap_debugfs_init,
 		.debugfs_cleanup = omap_debugfs_cleanup,
  
